pytorch_lightning/plugins/training_type/single_tpu.py

```diff
@@ -0,0 +1,34 @@
+import io
+from typing import Optional
+import torch
+from pytorch_lightning.plugins.training_type.single_device import SingleDevicePlugin
+from pytorch_lightning.utilities import _TPU_AVAILABLE, rank_zero_warn
+
+if _TPU_AVAILABLE:
+    import torch_xla
+    import torch_xla.core.xla_model as xm
+
+
+class SingleTPUPlugin(SingleDevicePlugin):
+    def __init__(self, device):
+        super().__init__(device)
+
+        self.tpu_local_core_rank = 0
+        self.tpu_global_core_rank = 0
+
+    def barrier(self, name: Optional[str] = None):
+        torch_xla.core.xla_model.rendezvous(f"pl.Trainer.{name}")
```

Comment:
Dont think we need to add rendezvous in `SingleTPUPlugin` as it is used to sync all processes. When using SingleTPU every process runs independently of the others.
https://github.com/pytorch/xla/issues/1976

Reply:
Will remove this!